src/Conference/ConferenceDetails.test.js

```diff
@@ -41,30 +46,48 @@ afterEach(() => {
 
 describe("ConferenceDetails", () => {
   const mockData = {
-    data: {
-      id: 1,
-      name: "Festival of Marketing",
-      dateTime: "2019-11-12T12:34:11Z",
-      city: "London",
-      description:
-        "From Festivalofmarketing.com: The Festival of Marketing is a unique experience where ambitious marketers can discover, learn, celebrate and shape the future together. As the largest global event dedicated to brand marketers, the Festival reflects the very nature of ...",
-      topic: "Marketing"
-    }
+    id: 1,
+    name: "Festival of Marketing",
+    dateTime: "2019-11-12T12:34:11Z",
+    city: "London",
+    description:
+      "From Festivalofmarketing.com: The Festival of Marketing is a unique experience where ambitious marketers can discover, learn, celebrate and shape the future together. As the largest global event dedicated to brand marketers, the Festival reflects the very nature of ...",
+    topic: "Marketing"
   };
 
-  it("renders conference details", done => {
-    expect.assertions(1);
+  it("renders conference details", async () => {
+    expect.assertions(6);
 
-    axios.get.mockResolvedValue(mockData);
+    const history = createMemoryHistory();
+    history.push("/1");
 
-    useParams.mockReturnValue({ id: "1" });
+    const apiReturnValue = Promise.resolve(mockData);
+    getConferenceById.mockImplementation(() => apiReturnValue);
 
-    const component = mount(<ConferenceDetails />);
+    let wrapper;
+    await act(async () => {
+      wrapper = mount(
+        <Router history={history}>
+          <ConferenceDetails />
+        </Router>
+      );
 
-    setImmediate(() => {
-      component.update();
-      expect(component).toMatchSnapshot();
-      done();
+      await apiReturnValue;
     });
+    wrapper.update();
+    expect(wrapper.containsMatchingElement(<h3>{mockData.name}</h3>)).toEqual(
```

Comment:
I think a better way of testing this is to add classnames to the html elements. Then you can find those in wrapper ie `expect(wrapper.find(className).length).toBe(1)` you can also check the text inside the element to match what you expect